Classes/Private/Manager/Objects/OCTActiveFile.m

```diff
@@ -0,0 +1,530 @@
+        // theObject.fileNumber = 0;
+        theObject.filePosition = 0;
+        theObject.filePath = fd;
+    }];
+    [[self.fileManager.dataSource managerGetRealmManager] noteMessageFileChanged:file];
+}
+
+- (void)markFileAsPaused:(OCTMessageFile *)file withFlags:(OCTPauseFlags)flag
+{
+    DDLogDebug(@"Marking file %@ as PAUSED.", file);
+
+    DDLogDebug(@"archiving resume data; if we die we're starting at offset %lld", self.bytesMoved);
+
+    NSData *conduitData = nil;
+    if ([self.conduit conformsToProtocol:@protocol(NSCoding)]) {
+        conduitData = [NSKeyedArchiver archivedDataWithRootObject:self.conduit];
+    }
+
+    [[self.fileManager.dataSource managerGetRealmManager] updateObject:file withBlock:^(OCTMessageFile *theObject) {
+        theObject.fileState = OCTMessageFileStatePaused;
+        theObject.pauseFlags = flag;
+        theObject.filePosition = self.bytesMoved;
+        theObject.restorationTag = conduitData;
+    }];
+    [[self.fileManager.dataSource managerGetRealmManager] noteMessageFileChanged:file];
+}
+
+- (void)markFileAsInterrupted:(OCTMessageFile *)file
+{
+    DDLogDebug(@"Marking file %@ as INTERRUPTED.", file);
+    DDLogDebug(@"archiving resume data; if we die we're starting at offset %lld", self.bytesMoved);
+
+    NSData *conduitData = nil;
+    if ([self.conduit conformsToProtocol:@protocol(NSCoding)]) {
+        conduitData = [NSKeyedArchiver archivedDataWithRootObject:self.conduit];
+    }
+
+    [[self.fileManager.dataSource managerGetRealmManager] updateObject:file withBlock:^(OCTMessageFile *theObject) {
+        theObject.fileState = OCTMessageFileStateInterrupted;
+        theObject.filePosition = self.bytesMoved;
+        theObject.restorationTag = conduitData;
+    }];
+    [[self.fileManager.dataSource managerGetRealmManager] noteMessageFileChanged:file];
+}
+
+- (void)resumeFile:(OCTMessageFile *)file
+{
+    DDLogDebug(@"archiving resume data; if we die we're starting at offset %lld", self.bytesMoved);
+
+    NSData *conduitData = nil;
+    if ([self.conduit conformsToProtocol:@protocol(NSCoding)]) {
+        conduitData = [NSKeyedArchiver archivedDataWithRootObject:self.conduit];
+    }
+
+    [[self.fileManager.dataSource managerGetRealmManager] updateObject:file withBlock:^(OCTMessageFile *theObject) {
+        theObject.fileState = OCTMessageFileStateLoading;
+        theObject.pauseFlags = OCTPauseFlagsNobody;
+        theObject.filePosition = self.bytesMoved;
+        theObject.restorationTag = conduitData;
+    }];
+    [[self.fileManager.dataSource managerGetRealmManager] noteMessageFileChanged:file];
+}
+
+#pragma mark - Public API - Controls
+
+- (BOOL)resumeWithError:(NSError *__autoreleasing *)error
+{
+    BOOL ok = YES;
+    if (error) {
+        *error = nil;
+    }
+
+    if ((self.fileMessage.fileState != OCTMessageFileStatePaused) &&
+        (self.fileMessage.fileState != OCTMessageFileStateWaitingConfirmation) ) {
+        // TODO make OCTFileError a constant symbol
```

Comment:
Will you do that? It would be add enum with error codes to `OCTManagerConstants`.